src/plugins/notebook/utils/notebook-entries.js

```diff
@@ -95,7 +95,8 @@ export const createNewEmbed = (snapshotMeta, snapshot = '') => {
         id: 'embed-' + date,
         name,
         snapshot,
-        type
+        type,
+        objectPath
```

Comment:
adding this causing error,
`Error in event handler for "updateEntries": "TypeError: Converting circular structure to JSON`

may be can do something like, `objectPath: JSON.stringify(objectPath)` then retrieve with `JSON.parse(self.embed.objectPath)`

Also worried about storing this objectPath info in each embed, overall performance for fetch, sort and search.
Let's discuss this tomorrow?

Reply:
In order for preview to work, we have to give it the correct objectPath. Using the url objectPath will cause all kinds of issues with other context menu actions in preview.

I turned it back to json before passing it on.